File: test/jvm/com/cattailsw/nanidroid/GhostPresentationFrameTest.java
package com.cattailsw.nanidroid;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

import org.junit.Test;

/** Characterizes the UI-free visibility policy used by presentation renderers. */
public class GhostPresentationFrameTest {
    @Test
    public void retainsLegacyBalloonVisibilityPolicyForBothSpeakers() {
        GhostPresentationFrame.Speaker hidden =
                new GhostPresentationFrame.Speaker("", "0", null, "-1");
        GhostPresentationFrame.Speaker visibleFromBalloon =
                new GhostPresentationFrame.Speaker("", "10", null, "0");
        GhostPresentationFrame.Speaker visibleFromText =
                new GhostPresentationFrame.Speaker("hello", "10", null, "-1");

        assertFalse(hidden.balloonVisible);
        assertTrue(visibleFromBalloon.balloonVisible);
        assertTrue(visibleFromText.balloonVisible);
    }
}

File: src/test/java/com/cattailsw/nanidroid/GhostPresentationFrameTest.kt
package com.cattailsw.nanidroid

import org.junit.Assert
import org.junit.Test

/** Characterizes the UI-free visibility policy used by presentation renderers.  */
class GhostPresentationFrameTest {
    @Test
    fun retainsLegacyBalloonVisibilityPolicyForBothSpeakers() {
        val hidden: GhostPresentationFrame.Speaker =
            GhostPresentationFrame.Speaker("", "0", null, "-1")
        val visibleFromBalloon: GhostPresentationFrame.Speaker =
            GhostPresentationFrame.Speaker("", "10", null, "0")
        val visibleFromText: GhostPresentationFrame.Speaker =
            GhostPresentationFrame.Speaker("hello", "10", null, "-1")

        Assert.assertFalse(hidden.balloonVisible)
        Assert.assertTrue(visibleFromBalloon.balloonVisible)
        Assert.assertTrue(visibleFromText.balloonVisible)
    }
}
